diff -Naur codesys-3_orig/Components/SysGraphicWindowQt/tvlineedit.cpp codesys-3/Components/SysGraphicWindowQt/tvlineedit.cpp
--- codesys-3_orig/Components/SysGraphicWindowQt/tvlineedit.cpp	2022-04-27 11:19:49.273536000 +0200
+++ codesys-3/Components/SysGraphicWindowQt/tvlineedit.cpp	2022-06-20 08:47:48.926234000 +0200
@@ -1,4 +1,8 @@
 #include "tvlineedit.h"
+#include <fcntl.h>
+#include <unistd.h>
+#include <QMutex>
+#include <QMutexLocker>
 #include "windowwidget.h"
 
 /* TVLineEdit implementation ----------------------------- */
@@ -67,6 +71,30 @@
 	parentWidget()->activateWindow();
 }
 
+static void openvk() {
+  //open virtual keyboard
+  int fd = open("/dev/virtualkeyboard", O_RDWR | O_NONBLOCK);
+  if (fd >= 0)
+  {
+    char * pCmd = "tvopen\n";
+    write(fd, pCmd, strlen(pCmd));
+    close(fd);
+  }
+}
+
+static QMutex vkmutex;
+void TVLineEdit::focusInEvent(QFocusEvent *event){
+  //open virtual keyboard
+  QMutexLocker locker(&vkmutex);
+  openvk();
+}
+
+void TVLineEdit::enterEvent(QEvent *event){
+	//open virtual keyboard
+	QMutexLocker locker(&vkmutex);
+	openvk();
+}
+
 bool TVLineEdit::isInEditControl(int iX, int iY)
 {
 	int x1, x2, y1, y2;
diff -Naur codesys-3_orig/Components/SysGraphicWindowQt/tvlineedit.h codesys-3/Components/SysGraphicWindowQt/tvlineedit.h
--- codesys-3_orig/Components/SysGraphicWindowQt/tvlineedit.h	2022-04-27 11:19:49.273536000 +0200
+++ codesys-3/Components/SysGraphicWindowQt/tvlineedit.h	2022-06-20 08:44:51.158703000 +0200
@@ -44,4 +44,8 @@
 private:
     //Position:
     int m_left, m_top, m_right, m_bottom;
+    
+protected:
+	void focusInEvent(QFocusEvent *event);
+	void enterEvent(QEvent *event);  
 };
