From 20c9dc02d576aa478069e67dacc51328b1b1ac07 Mon Sep 17 00:00:00 2001
From: Oleg Karfich <oleg.karfich@wago.com>
Date: Tue, 14 Jan 2020 10:55:53 +0100
Subject: [PATCH] add new version w02.06.01-vtpctp

Signed-off-by: Oleg Karfich <oleg.karfich@wago.com>
---
 localversion-wago | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-wago b/localversion-wago
index 110d663..5e89971 100644
--- a/localversion-wago
+++ b/localversion-wago
@@ -1 +1 @@
-w02.06.00-vtpctp
+w02.06.01-vtpctp
-- 
2.7.4

