diff -Naur qt-everywhere-opensource-src-5.9.2_org/qtbase/configure.pri qt-everywhere-opensource-src-5.9.2/qtbase/configure.pri
--- qt-everywhere-opensource-src-5.9.2_org/qtbase/configure.pri	2017-10-02 10:43:38.000000000 +0200
+++ qt-everywhere-opensource-src-5.9.2/qtbase/configure.pri	2017-10-20 11:14:01.000000000 +0200
@@ -316,6 +316,10 @@
         # cross compiling, check that pkg-config is set up sanely
         sysroot = $$config.input.sysroot
 
+        isEmpty(sysroot) {
+          sysroot = $$getenv("PKG_CONFIG_SYSROOT_DIR")
+        }
+
         pkgConfigLibdir = $$getenv("PKG_CONFIG_LIBDIR")
         isEmpty(pkgConfigLibdir) {
             isEmpty(sysroot) {
