diff -Naur qt-everywhere-opensource-src-5.9.2_org/qtwebengine/src/core/config/linux.pri qt-everywhere-opensource-src-5.9.2/qtwebengine/src/core/config/linux.pri
--- qt-everywhere-opensource-src-5.9.2_org/qtwebengine/src/core/config/linux.pri	2017-10-27 08:30:02.000000000 +0200
+++ qt-everywhere-opensource-src-5.9.2/qtwebengine/src/core/config/linux.pri	2017-11-03 10:16:41.821080570 +0100
@@ -63,7 +63,7 @@
     gn_args += sysroot=\"$${PKG_CONFIG_SYSROOT_DIR}\"
 
     !lessThan(MARMV, 8) {
-        gn_args += arm_use_neon=true
+        gn_args += arm_use_neon=false
     } else {
         MFPU = $$extractCFlag("-mfpu=.*")
         !isEmpty(MFPU):contains(MFPU, ".*neon.*") {
@@ -72,7 +72,7 @@
             gn_args += arm_use_neon=false
             # If the toolchain does not explicitly specify to use NEON instructions
             # we use arm_neon_optional for ARMv7
-            equals(MARMV, 7): gn_args += arm_optionally_use_neon=true
+            equals(MARMV, 7): gn_args += arm_optionally_use_neon=false
         }
     }
 
