diff -Naur qt-everywhere-opensource-src-5.9.2_org/qtbase/src/gui/kernel/qguiapplication.cpp qt-everywhere-opensource-src-5.9.2/qtbase/src/gui/kernel/qguiapplication.cpp
--- qt-everywhere-opensource-src-5.9.2_org/qtbase/src/gui/kernel/qguiapplication.cpp	2017-11-28 10:03:53.000000000 +0100
+++ qt-everywhere-opensource-src-5.9.2/qtbase/src/gui/kernel/qguiapplication.cpp	2017-11-28 16:15:55.156986703 +0100
@@ -2450,6 +2450,7 @@
 {
     QGuiApplicationPrivate *d = self;
     modifier_buttons = e->modifiers;
+    static int iFirstTouch = 0;
 
     if (e->touchType == QEvent::TouchCancel) {
         // The touch sequence has been canceled (e.g. by the compositor).
@@ -2647,6 +2648,21 @@
             continue;
         }
 
+        if (qEnvironmentVariableIsSet("QT_TOUCH_SUPPRESS_SECOND_TOUCH_BEGIN")) {
+          if (eventType == QEvent::TouchBegin) {
+            if(iFirstTouch == 0) {
+              //qDebug() << "### --> qtguiapplication.cpp --> first Touch !!! QEvent::TouchBegin --> QGuiApplicationPrivate::processTouchEvent";
+              iFirstTouch++;
+            } else {
+              //qDebug() << "### --> qtguiapplication.cpp --> second Touch supressed !!! QEvent::TouchBegin -> xi2HandleEvent";
+              iFirstTouch = 0;
+              continue;
+            }
+          } else {
+            iFirstTouch = 0;
+          }
+        }
+
         QTouchEvent touchEvent(eventType,
                                e->device,
                                e->modifiers,
