--- qt-everywhere-opensource-src-5.9.2_org/qtwebengine/src/3rdparty/chromium/content/browser/renderer_host/input/touch_action_filter.cc	2017-09-19 18:15:15.000000000 +0200
+++ qt-everywhere-opensource-src-5.9.2/qtwebengine/src/3rdparty/chromium/content/browser/renderer_host/input/touch_action_filter.cc	2018-01-19 14:37:56.860913610 +0100
@@ -6,12 +6,33 @@
 
 #include <math.h>
 
+#include <unistd.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+
 #include "base/logging.h"
 #include "third_party/WebKit/public/platform/WebGestureEvent.h"
 
 using blink::WebInputEvent;
 using blink::WebGestureEvent;
 
+static bool bDropPinch = false;
+
+int fileexists(char * pFilename)
+// Check if file is available
+{
+  struct stat sts;
+  if (stat(pFilename, &sts) == -1 && errno == ENOENT)
+  {
+    return -1;
+  }
+  else
+  {   
+    return 0;
+  }
+}
+
 namespace content {
 namespace {
 
@@ -29,10 +50,15 @@
 
 TouchActionFilter::TouchActionFilter() :
   drop_scroll_gesture_events_(false),
-  drop_pinch_gesture_events_(false),
   drop_current_tap_ending_event_(false),
   allow_current_double_tap_event_(true),
   allowed_touch_action_(TOUCH_ACTION_AUTO) {
+    
+  char * pFile = "/tmp/custom_disable_pinch.txt";
+  if (fileexists(pFile) == 0){
+    bDropPinch = true;
+  }
+  drop_pinch_gesture_events_ = bDropPinch;
 }
 
 bool TouchActionFilter::FilterGestureEvent(WebGestureEvent* gesture_event) {
@@ -45,8 +71,15 @@
     case WebInputEvent::GestureScrollBegin:
       DCHECK(!drop_scroll_gesture_events_);
       DCHECK(!drop_pinch_gesture_events_);
-      drop_pinch_gesture_events_ =
-          (allowed_touch_action_ & TOUCH_ACTION_PINCH_ZOOM) == 0;
+      if (bDropPinch)
+      {
+        drop_pinch_gesture_events_ = true;
+      }
+      else
+      {
+        drop_pinch_gesture_events_ =
+            (allowed_touch_action_ & TOUCH_ACTION_PINCH_ZOOM) == 0;
+      }
       drop_scroll_gesture_events_ = ShouldSuppressScroll(*gesture_event);
       return drop_scroll_gesture_events_;
 
@@ -97,7 +130,7 @@
 
     case WebInputEvent::GesturePinchEnd:
       if (drop_pinch_gesture_events_) {
-        drop_pinch_gesture_events_ = false;
+        drop_pinch_gesture_events_ = bDropPinch;
         return true;
       }
       break;
@@ -147,7 +180,7 @@
 }
 
 bool TouchActionFilter::FilterScrollEndingGesture() {
-  drop_pinch_gesture_events_ = false;
+  drop_pinch_gesture_events_ = bDropPinch;
   if (drop_scroll_gesture_events_) {
     drop_scroll_gesture_events_ = false;
     return true;
