diff -uNr xf86-input-tslib-0.0.6_org/src/tslib.c xf86-input-tslib-0.0.6/src/tslib.c
--- xf86-input-tslib-0.0.6_org/src/tslib.c	2014-04-07 09:51:37.000000000 +0200
+++ xf86-input-tslib-0.0.6/src/tslib.c	2014-04-07 09:54:34.424675557 +0200
@@ -194,10 +194,6 @@
 			x = samp.x;
 			y = samp.y;
 
-			xf86XInputSetScreen(local, priv->screen_num,
-					samp.x,
-					samp.y);
-
 			xf86PostMotionEvent (local->dev, TRUE, 0, 2,
 					x, y);
 
@@ -282,12 +278,6 @@
 				break;
 		}
 	}
-
-	if (ret < 0) {
-		ErrorF("ts_read failed\n");
-		return;
-	}
-
 }
 
 #if GET_ABI_MAJOR(ABI_XINPUT_VERSION) >= 7
