From: Florian Seidel <florian.seidel@wago.com>
Date: Thu, 12 Dec 2024 09:52:35 +0100
Subject: [PATCH] do not remove host header from http request
---
 src/pki/scep/scep.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/pki/scep/scep.c b/src/pki/scep/scep.c
index 4684774bb7b2..3d6d0027b131 100644
--- a/src/pki/scep/scep.c
+++ b/src/pki/scep/scep.c
@@ -364,7 +364,6 @@ bool scep_http_request(const char *url, scep_op_t op, bool http_post,
 				status = lib->fetcher->fetch(lib->fetcher, complete_url, response,
 										 FETCH_TIMEOUT, http_timeout,
 										 FETCH_REQUEST_HEADER, "Pragma:",
-										 FETCH_REQUEST_HEADER, "Host:",
 										 FETCH_REQUEST_HEADER, "Accept:",
 										 FETCH_SOURCEIP, srcip,
 										 FETCH_RESPONSE_CODE, http_code,
