--- codesys-3_org/Components/CmpTargetVisu/CmpTargetVisuImpl.c	2019-04-26 11:16:42.630542999 +0200
+++ codesys-3/Components/CmpTargetVisu/CmpTargetVisuImpl.c	2019-04-26 08:31:42.829440727 +0200
@@ -5,6 +5,9 @@
 #include "CmpTargetVisuImpl.h"
 #include "CmpTargetVisuBmpBuffer.h"
 
+#include <fcntl.h>
+#include <unistd.h>
+
 #define CMP_VENDORID	RTS_VENDORID_3S
 
 #define SYSGRAPHIC_TRANSFORMATION_SUPPORTED (CHK_SysGraphicCreateTransformation && CHK_SysGraphicFreeTransformation && \
@@ -4220,6 +4223,14 @@
 	{
 		CAL_SysMemFreeData(COMPONENT_NAME, pCmd->pwsInitVal);
 	}
+ 
+  int fd = open("/dev/virtualkeyboard", O_RDWR | O_NONBLOCK);
+  if (fd >= 0)
+  {
+    char * pCmd = "tvopen\n";
+    write(fd, pCmd, strlen(pCmd));
+    close(fd);
+  }
 }
 
 static void PaintOpenEditControl(void* pData, TargetvisuParams* pParamData, RTS_UI32 dwCmdSize, RTS_BOOL bSwapping)
