diff -Naur udev-182_org/rules/50-udev-default.rules udev-182/rules/50-udev-default.rules
--- udev-182_org/rules/50-udev-default.rules	2012-01-29 01:15:46.000000000 +0100
+++ udev-182/rules/50-udev-default.rules	2017-03-21 16:19:11.906503501 +0100
@@ -20,7 +20,7 @@
 KERNEL=="mem|kmem|port|nvram", GROUP="kmem", MODE="0640"
 
 # input
-SUBSYSTEM=="input", ENV{ID_INPUT}=="", IMPORT{builtin}="input_id"
+ENV{SUBSYSTEM}="input", IMPORT{builtin}="input_id %p"
 KERNEL=="mouse*|mice|event*", MODE="0640"
 KERNEL=="ts[0-9]*|uinput", MODE="0640"
 KERNEL=="js[0-9]*", MODE="0644"
